Webpack 5 Support / Refactoring

web-mi opened the pull request:
I have refactored the code a little bit and added support for webpack 5 but webpack 4 is still be supported.
Maybe it should be released as major but it should not include any breaking changes.

- Webpack 5 support
- Refactored logic from prototype class to class
- Switched from done hook to afterEmit because webpack v5 shrinks assets to SizeOnlySource [Changelog](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#sizeonlysource-after-emit)
- Added config validation
- Restructured options from single attributes to options attribute

Fixes #63

statianzo commented on index.js:
```diff
@@ -1,175 +1,249 @@
+        portfinder.basePort = this.defaultPort;
+        portfinder.getPort((err, port) => {
+          if (err) {
+            throw err;
```
Not your change, but I noticed this while reviewing. For playing nice as a plugin, this error should be sent up through the callback instead of throwing.

```suggestion
            return cb(err);
```

statianzo commented on index.js:
```diff
@@ -1,175 +1,249 @@
+
+    const sourceHash = LiveReloadPlugin.generateHashCode(data[1].source());
+
+    if (this.sourceHashs?.[data[0]] === sourceHash) {
```
`this.sourceHashes` should always be defined so optional chaining shouldn't be needed. Removing also prevents a syntax error in node v12

```suggestion
    if (this.sourceHashs.[data[0]] === sourceHash) {
```

statianzo reviewed (commented):
This looks great! Webpack 5 and some much needed modernization. :tada: 

Thank you @web-mi :pray: 

It's worth cutting a new major because of the options validation and modern ES syntax.

web-mi commented:
@statianzo could you tell me why you unset delay option for tiny-lr?
I could not find any usage of delay in tiny-lr but won't remove it because i don't know why you removed it.

web-mi commented:
Changes are done.
I refactored the listen function a little bit.
The package.json now includes the peerDependency for webpack and engines for node version.
And i have added a changelog file.

statianzo commented:
> @statianzo could you tell me why you unset delay option for tiny-lr?
> I could not find any usage of delay in tiny-lr but won't remove it because i don't know why you removed it.

It came as part of #38. Likely being conservative about what gets passed through to tiny-lr, to prevent issues if it were to add `delay`.

statianzo commented:
Released v3.0.0. Switched to github actions for CI because travis seems to no longer run.

Thanks again @web-mi
